device/core/src/client.ts

```diff
@@ -35,6 +42,12 @@ export class Client extends EventEmitter {
    */
   static sasRenewalInterval: number = 2700000;
 
+  /**
+   * Maximum timeout (in milliseconds) used to consider an operation failed.
+   * The operation will be retried according to the retry policy set with {@link azure-iot-device.Client.setRetryPolicy} method (or {@link azure-iot-common.ExponentialBackoffWithJitter} by default) until this value is reached.)
+   */
+  maxOperationTimeout: number;
```

Comment:
>maxOperationTimeout [](start = 2, length = 19)

I don't see anywhere for customers to change this.  Are we adding an API later?

Reply:
i've been going back and forth on this. technically it's a public property. I think having a method that acts as a setter is overkill since it's just a number.  I'm also not sure other SDKs let customers customize this. so right now I'm going with a minimalist approach. I'm open to feedback though. 

---
In reply to: [145965088](https://github.com/Azure/azure-iot-sdk-node/pull/114#discussion_r145965088) [](ancestors = 145965088)

Reply:
maybe we should make it private for now. also I need to double-check what the default value really should be.

---
In reply to: [146043848](https://github.com/Azure/azure-iot-sdk-node/pull/114#discussion_r146043848) [](ancestors = 146043848,145965088)

Reply:
I'm going to make it private for now - official default value is 4 minutes. if we need to make it public later we'll find a way.

---
In reply to: [146043956](https://github.com/Azure/azure-iot-sdk-node/pull/114#discussion_r146043956) [](ancestors = 146043956,146043848,145965088)